### PR TITLE
add  chapter on relations from mastering plone (slighty updated)

### PR DESCRIPTION
**Pull request by pbauer**

### CODE REVIEW

**Review by ksuess — COMMENTED** [review]
From time to time I think it would be good to stop defining target sets of a relation field in its widget. Cause this is a constraint of its data structure (Plone: schema) and not a matter of the display (Plone: widget).

File: docs/backend/relations.md
Comment (ksuess) [inline docs/backend/relations.md]: This is some how astonishing to motivate a switch to new `StaticCatalogVocabulary` with display decision. Why not start this chapter with `StaticCatalogVocabulary`. Later the traditional way can be mentioned, to understand existing code.

Comment (pbauer) [reply docs/backend/relations.md]: > From time to time I think it would be good to stop defining target sets of a relation field in its widget. Cause this is a constraint of its data structure (Plone: schema) and not a matter of the display (Plone: widget).

I updated the examples where the catalog-query was configures twice. It turns out that is not needed.

Comment (pbauer) [reply docs/backend/relations.md]: `StaticCatalogVocabulary` only works with widgets other than `RelatedItemsFieldWidget`. So the default is still that and only if you require other widgets you can use `StaticCatalogVocabulary`.

**Comment by stevepiercy** [comment]
I will review this in the next day or two. I'm working on other docs at the moment. Please do not merge yet. Thank you!

**Comment by stevepiercy** [comment]
General comment, with details coming in a full review later.

Documentation should be the authoritative source, not Training. We should pull content from the Training, and insert and adapt it in Documentation.

If that is too much work for now, then I would be OK with leaving the `seealso` directives on the previously empty pages (for example, `docs/backend/behaviors.md`) because it is an improvement, giving the reader something to use as a reference. We should also add a `todo` directive for each of these pages, and create a new issue in the Documentation issue tracker for each one.

I also think that this would give some newbies an opportunity to contribute by repurposing Training material into Documentation. Or so I hope.

@pbauer what do you think?

File: docs/backend/control-panels.md
Comment (stevepiercy) [inline docs/backend/control-panels.md]: This link is broken. See rendered:

https://deploy-preview-1290--6-dev-docs-plone-org.netlify.app/backend/control-panels.html

Is this the page you want to link to?

https://training.plone.org/5/mastering-plone/registry.html#vocabularies-registry-settings-and-control-panels

File: docs/backend/vocabularies.md
Comment (stevepiercy) [inline docs/backend/vocabularies.md]: This link is broken, too.

https://deploy-preview-1290--6-dev-docs-plone-org.netlify.app/backend/vocabularies.html

Is this the link you want?

https://training.plone.org/5/mastering-plone/registry.html#vocabularies

File: docs/backend/widgets.md
Comment (stevepiercy) [inline docs/backend/widgets.md:21]: This link goes to the top of this page, instead of an anchor in the page that is more relevant:

https://training.plone.org/5/mastering-plone/dexterity_reference.html#widgets

However that only links to the Plone 5.2 docs, which is not a great idea.

https://docs.plone.org/external/plone.app.dexterity/docs/reference/widgets.html

But that brings up another larger question. Should we pull the Dexterity docs into Documentation?

https://github.com/plone/plone.app.dexterity/tree/master/docs

**Review by stevepiercy — CHANGES_REQUESTED** [review]
This is my first round of review.

I did not start on the mammoth `backend/relations.md`, but I will start nibbling on that pachyderm one bite at a time over the next day or so.

File: docs/backend/control-panels.md
Comment (pbauer) [reply docs/backend/control-panels.md]: This only happens because the deployment of the training is currently broken because of links that are supposedly broken (they are not). I'll fix that.

File: docs/backend/widgets.md
Comment (pbauer) [reply docs/backend/widgets.md:21]: The Dexterity-reference from the training (https://training.plone.org/5/mastering-plone/dexterity_reference.html) and the dexterity docs should both move into docs. I would not pull the dexterity-docs in (as a submodule) but rather move it there since it is very very "core".

File: docs/backend/control-panels.md
Comment (pbauer) [reply docs/backend/control-panels.md]: Fixed the training deploy. Now these links should start working.

File: docs/backend/vocabularies.md
Comment (pbauer) [reply docs/backend/vocabularies.md]: Same

**Comment by pbauer** [comment]
> Documentation should be the authoritative source, not Training. We should pull content from the Training, and insert and adapt it in Documentation.

Yes. The chapters on relations, and the dexterity-reference can be moved to the docs without mayor changes. I'll remove them from the training and link to the docs once these are merged. 

> If that is too much work for now, then I would be OK with leaving the `seealso` directives on the previously empty pages (for example, `docs/backend/behaviors.md`) because it is an improvement, giving the reader something to use as a reference. We should also add a `todo` directive for each of these pages, and create a new issue in the Documentation issue tracker for each one.

Yes
 
> I also think that this would give some newbies an opportunity to contribute by repurposing Training material into Documentation. Or so I hope.

You can always hope but I doubt it.

File: docs/backend/control-panels.md
Comment (stevepiercy) [reply docs/backend/control-panels.md]: The link is still broken: https://deploy-preview-1290--6-dev-docs-plone-org.netlify.app/backend/control-panels.html

Screen Shot 2022-09-07 at 5 26 34 AM

File: docs/backend/widgets.md
Comment (stevepiercy) [reply docs/backend/widgets.md:21]: Issue created: https://github.com/plone/documentation/issues/1298

Resolving this comment.

File: docs/backend/vocabularies.md
Comment (stevepiercy) [reply docs/backend/vocabularies.md]: Still broken. See screenshot.

Screen Shot 2022-09-07 at 5 26 05 AM

File: docs/backend/control-panels.md
Comment (stevepiercy) [reply docs/backend/control-panels.md]: I retried a Netlfiy preview deploy, as it hadn't been updated in a couple of days. The link now renders correctly. Resolving.

File: docs/backend/vocabularies.md
Comment (stevepiercy) [reply docs/backend/vocabularies.md]: Rebuild fixed the link. Resolving.

**Comment by stevepiercy** [comment]
Issues created:

- #1302
- #1303
- #1304 
- #1305 
- #1306 
- #1307 
- #1308

**Comment by stevepiercy** [comment]
Added meta data and TODOs in [443a010](https://github.com/plone/documentation/pull/1290/commits/443a01045af1d3f826436bc6f31c1e98768b01ab).

**Comment by stevepiercy** [comment]
I'm going to continue editing and pushing commits to this tomorrow. The commits I pushed so far are just meta and technical stuff that we should include in docs.

File: docs/backend/relations.md
Comment (stevepiercy) [inline docs/backend/relations.md]: Line 18 does not exist in the [rendered code example](https://deploy-preview-1290--6-dev-docs-plone-org.netlify.app/backend/relations.html#using-different-widgets-for-relations). Did you instead mean these lines?

```suggestion
:emphasize-lines: 1, 8
```

File: docs/backend/relations.md
Comment (stevepiercy) [inline docs/backend/relations.md]: The code appears to include a relationship to both Documents and Events. Can you verify?

Also is it a single or multiple choice select?

File: docs/backend/relations.md
Comment (stevepiercy) [inline docs/backend/relations.md]: Not valid syntax, and missing a value for `basePath`.

```suggestion
        "basePath": "/",
```

File: docs/backend/relations.md
Comment (stevepiercy) [inline docs/backend/relations.md:207]: These are the only two instances of `RelationWidget` throughout the docs. Also the image itself says "Relationlist Field in Search Mode". Would you please verify that the alt text and caption are correct?

File: docs/backend/relations.md
Comment (stevepiercy) [inline docs/backend/relations.md]: There is a reference in the code to `RelatedItemsFieldWidget`. Is this term "Related Items Widget" a shortening of its code name?

File: docs/backend/relations.md
Comment (stevepiercy) [inline docs/backend/relations.md]: There is a reference in the code to `RelatedItemsFieldWidget`. Is this term "Related Items Widget" a shortening of its code name?

File: docs/backend/relations.md
Comment (stevepiercy) [inline docs/backend/relations.md:374]: The above alt text and caption do not align with the code or the text in the image, "RelationChoice with Select Widget". Is this correct?

**Review by stevepiercy — CHANGES_REQUESTED** [review]
This is excellent work. To save review cycles, I made lots of nit-picky commits to clean up English and MyST syntax and grammar.

I have a few additional questions. Please see the comments and suggestions.

Thank you!

File: docs/backend/relations.md
Comment (pbauer) [reply docs/backend/relations.md:207]: They are :)

File: docs/backend/relations.md
Comment (pbauer) [reply docs/backend/relations.md]: The python-class is called `RelatedItemsFieldWidget` for several reasons. A good human readable term is probably Relation-Widget (i.e. default widget for realtion-fields).

Comment (pbauer) [reply docs/backend/relations.md]: I decided on lower-case: relations-widget

File: docs/backend/relations.md
Comment (pbauer) [reply docs/backend/relations.md:374]: Fixed it

File: docs/backend/relations.md
Comment (pbauer) [reply docs/backend/relations.md]: fixed it

File: docs/backend/relations.md
Comment (pbauer) [reply docs/backend/relations.md]: fixed

File: docs/backend/relations.md
Comment (stevepiercy) [reply docs/backend/relations.md]: English does not use hyphenation in such terms. Does German hyphenate a lot?

Also is it better to use singular or plural, `relation widget` vs. `relations widget`?

Comment (pbauer) [reply docs/backend/relations.md]: germans love to hyphenate, we are word-hyphenators. I would prefer plural (relations widget).

**Pull request merged**
